data_steward/analytics/cdr_ops/rdr_export_qc.py

```diff
@@ -327,33 +343,100 @@
  ,concept_class_id
  ,n
 FROM ppi_concept_code
-JOIN `{project_id}.{new_rdr}.concept`
+JOIN `{{project_id}}.{{new_rdr}}.concept`
  ON LOWER(concept_code)=LOWER(code)
 WHERE LOWER(concept_class_id)<>LOWER(expected_concept_class_id)
 AND CASE WHEN expected_concept_class_id = 'Question' THEN concept_class_id NOT IN('Topic','PPI Modifier') END
 AND concept_class_id != 'Qualifier Value'
 ORDER BY 1, 2, 3
-'''
+''')
+query = tpl.render(new_rdr=new_rdr, project_id=project_id)
 execute(client, query)
 
-# # Make sure smoking data still exists
+# # Identify Questions That Dont Exist in the RDR Export
+# This identifies questions as indicated by a PPI vocabulary and Question concept_class_id that
+# do not exist in the dataset.
+
+tpl = JINJA_ENV.from_string("""
+with question_codes as (select c.concept_id, c.concept_name, c.concept_class_id
+from `{{project_id}}.{{new_rdr}}.concept` as c
+where REGEXP_CONTAINS(c.vocabulary_id, r'(?i)(ppi)') and REGEXP_CONTAINS(c.concept_class_id, r'(?i)(question)'))
+, used_q_codes as (
+    select distinct o.observation_source_concept_id, o.observation_source_value
+    from `{{project_id}}.{{new_rdr}}.observation` as o
+    join `{{project_id}}.{{new_rdr}}.concept` as c
+    on o.observation_source_concept_id = c.concept_id
+    where REGEXP_CONTAINS(c.vocabulary_id, r'(?i)(ppi)') and REGEXP_CONTAINS(c.concept_class_id, r'(?i)(question)')
+)
+    SELECT *
+    from question_codes
+    where concept_id not in (select observation_source_concept_id from used_q_codes)
+    """)
+query = tpl.render(new_rdr=new_rdr, project_id=project_id)
+execute(client, query)
+
+# # Identify Questions That Dont Exist in the Cleaned RDR Export
+# This identifies questions as indicated by a PPI vocabulary and Question concept_class_id that
+# do not exist in the cleaned dataset but did exist in the raw dataset.
+
+if raw_rdr == "default":
```

Comment:
I'm unfamiliar with this query, but I see that you're using the "default" guard here, so this is the meat of the change.  Can you provide some context why this doesn't need a test (it doesn't need to be an essay, I'd just like to understand it more being new)?

Reply:
We are running this using our jupyter notebook runner.  That runner looks at the parameters defined at the top of the file to identify required parameters to run the notebook.  This particular notebook could use further refinement.  As of now, it is being run on a raw RDR import and then again on a cleaned RDR import.  Because we can learn different things from the raw vs. cleaned data.  The test below is only valid if the data has been cleaned.  So, if `-p raw_rdr YYYYq#r#_rdr` is provided at the command line, then the "default" value is overridden and the query is run.  If not provided, the script doesn't break, the query just doesn't run.  I am checking against the value "default" because that is defined in `analytics/cdr_ops/report_runner.py` to identify optional notebook parameters.

The important thing about this query is it is trying to identify instances where entire questions have been dropped from the cleaned RDR dataset but are present in the raw RDR dataset.  If a question is dropped entirely, this indicates a problem in the RDR cleaning process.